### DISCUSSION TITLE
A-10C (II)  | CDU, COMMS, ILS/TCN

### DISCUSSION THREAD

**Discussion by Ayokas** [charliefoxtwo/TouchDCS#17] [Show and tell]
I just found this project and startet with a Control Panel for the A-10C (II). Especially CDU is a real pain if you have to look and click ;-)

Currently CDU is finished for testing and COMMS, ILS/TCN is scheduled to be done in the next few days.

![TouchOSC_20220117_200016](https://user-images.githubusercontent.com/14794113/149910180-983c9aef-5f40-4f38-b7ff-942298d2dbca.png)

**Comment by charliefoxtwo** [comment]
Nice! I actually did a CDU layout a while ago, it's been at 80% completion for over a year now (functionally complete, but looks terrible). Mine is also portrait-oriented, so it's great to see some variety. I like that you made the numpad bigger, the numpad on mine is "to-scale" and it kind of makes it a pain to easily enter coordinates. 

The circle with a dot in the middle and the block cursor (and a couple of other characters used on the cdu display) actually do exist in unicode if you can hunt them down. I'll see if I can dig up the character codes; I was able to find them for my layout.

**Reply by Ayokas** [reply]
Yea I tried the landscape mode, because it is scales better for the keyboard and so easier to operate. I value functionality a bit more than immersion so I broke the layout a bit.
Additionally I primarily input coordinates into my flight plan / waypoints so I need the numpad more often than the rest of the keyboard =)

I'm currently experimenting with the dials of the AAP and STEERPT to use ASCII symbols for the arrow on the dial. Will keep the progress posted in here

**Reply by charliefoxtwo** [reply]
This is what I used to remap characters in the CDU by the way. 

```lua
local map = 
{
  ["¶"] = "▓",
  ["®"] = "↕",
  ["©"] = "ʘ",
  ["¡"] = "[]",
  ["«"] = "←",
  ["»"] = "→",
}
```

**Comment by Ayokas** [comment]
It has been quite a while but I finally finished the ILS / TCN Page. Originally I wanted to use dials but they were just to flickery so I reverted back to buttons =)

I used the remapping of values to ASCII characters to change the arrow in the ILS-Switch.
```LUA
function onReceiveOSC(message, connections)
  local path = message[1]
  local arguments = message[2]
  
  for i=1,#arguments do
    if (arguments[i].tag == 'i') then
      if (arguments[i].value == 0) then
        self.values['text'] = '←'
      else
        self.values['text'] = '↑'
      end
    end
  end
end
```

Next up is COMMS page. Probably this will be finished between christmas and new year.
![Screenshot_2022-12-16-20-45-49-375](https://user-images.githubusercontent.com/14794113/208177426-e830caa5-d858-4125-8568-12b1ce9cd470.jpeg)

**Reply by charliefoxtwo** [reply]
Dials can be difficult to get working just right. For discrete selections (not sure if that's what you need here) I was able to work something up for the hind that you might be able to reuse (see the "Pods" selector) https://github.com/charliefoxtwo/TouchDCS-Layouts/tree/main/Mi-24P
